Wikipedia/Code/ReadingListsController.swift

```diff
@@ -452,7 +452,35 @@ public typealias ReadingListsController = WMFReadingListsController
             NotificationCenter.default.post(name: ReadingListsController.readingListsServerDidConfirmSyncWasEnabledForAccountNotification, object: nil, userInfo: [ReadingListsController.readingListsServerDidConfirmSyncWasEnabledForAccountWasSyncEnabledKey: NSNumber(value: wasSyncEnabledForAccount), ReadingListsController.readingListsServerDidConfirmSyncWasEnabledForAccountWasSyncEnabledOnDeviceKey: NSNumber(value: wasSyncEnabledOnDevice), ReadingListsController.readingListsServerDidConfirmSyncWasEnabledForAccountWasSyncDisabledOnDeviceKey: NSNumber(value: wasSyncDisabledOnDevice)])
         }
     }
-    
+
+    public func eraseAllSavedArticlesAndReadingLists() {
+        assert(Thread.isMainThread)
+
+        let oldSyncState = syncState
+        var newSyncState = oldSyncState
+
+        if isSyncEnabled {
+            // Since there is no batch delete on the server,
+            // we remove local and remote reading lists
+            // by disabling and then enabling the service.
+            // Otherwise, we'd have to delete everything via single requests.
+            newSyncState.insert(.needsRemoteDisable)
```

Comment:
does this need a local clear as well?

Reply:
My thinking was -

1. Service is disabled remotely (reading lists are removed on the server)
2. Service is enabled remotely
3. Sync (get lists from the server, delete local lists if they're not on the server - local lists get deleted)

If I add a local clear, it would happen after sync, in `processLocalUpdates`. And at that time, there should be no lists to clear (because they were deleted in `createOrUpdate`)